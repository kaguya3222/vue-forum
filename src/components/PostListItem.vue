<template>
  <div class="post">
    <div class="user-info">
      <a href="#" class="user-name">{{ user.name }}</a>

      <a href="#">
        <img class="avatar-large" :src="user.avatar" alt="" />
      </a>

      <p class="desktop-only text-small">{{ userPostsCount }} posts</p>
    </div>

    <div class="post-content">
      <div v-if="!editing">
        {{ post.text }}
      </div>
      <div class="col-full" v-else>
        <post-editor :post="post" @save="editing = false" />
      </div>
    </div>

    <app-date :unixDate="post.publishedAt" />
  </div>
</template>

<script>
import PostEditor from "./PostEditor";

import { countObjectProperties } from "../helpers";
import { mapGetters } from "vuex";

export default {
  name: "PostListItem.vue",
  components: {
    PostEditor
  },
  props: {
    post: {
      required: true,
      type: Object
    }
  },
  data() {
    return {
      editing: true
    };
  },
  computed: {
    ...mapGetters(["users"]),
    user() {
      return this.users[this.post.userId];
    },
    userPostsCount() {
      return countObjectProperties(this.user.posts);
    }
  }
};
</script>
