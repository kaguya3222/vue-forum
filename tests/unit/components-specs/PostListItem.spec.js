import { shallowMount } from "@vue/test-utils";
import PostListItem from "../../../src/components/PostListItem";
import mockedSourceData from "../mocks/mockedSourceData";

describe("PostListItem", () => {
  test("Shows post data", () => {
    const wrapper = shallowMount(PostListItem, {
      propsData: {
        post: mockedSourceData.posts["-KsjWehQ--apjDBwSBCZ"]
      },
      stubs: {
        "app-date": true
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
