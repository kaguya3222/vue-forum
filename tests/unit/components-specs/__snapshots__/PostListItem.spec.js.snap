// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostListItem Shows post data 1`] = `
<div class="post">
  <div class="user-info"><a href="#" class="user-name">Rolf</a> <a href="#"><img src="https://firebasestorage.googleapis.com/v0/b/forum-2a982.appspot.com/o/images%2Favatars%2Frah?alt=media&amp;token=7ad21914-a4f4-4ad0-add6-17e6d3ae9679" alt="" class="avatar-large"></a>
    <p class="desktop-only text-small">5 posts</p>
  </div>
  <div class="post-content">
    <div>
      I'm curious to know what kind of pets the forum users have. I thought this could be a nice way to get to know each other a little better, so I'll start!

      My dog is a five month old Italian mastiff (Cane Corso). Since I got him from Barcelona I thought a Spanish name would be suitable, and I named him Don Jefe, or Jefe for short (which means "boss man").

      Jefe is a cute, little beast who eats 1.5 kg raw meat every day. He love to fart, snore and play.

      Here's Jefe with his Instagram hat and sqeaky ball ðŸŽ‰ðŸŽ‰ðŸŽ‰
      ![Don Jefe](http://i.imgur.com/XCWzR6l.jpg)

    </div>
  </div>
  <app-date-stub unixdate="1504035908"></app-date-stub>
</div>
`;
